jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -0,0 +1,6134 @@
+	 *  is illegal, or if size is invalid
+	 */
+	public void copyMemory(long srcAddress, long destAddress, long size) {
+		copyMemory(null, srcAddress, null, destAddress, size);
+	}
+
+	/**
+	 * Swap specified bytes between source object and destination
+	 * 
+	 * @param srcObj object to copy from 
+	 * @param srcOffset location in srcObj to start swap
+	 * @param destObj object to copy into
+	 * @param destOffset location in destObj to start swap
+	 * @param srcSize the number of bytes to be swapped in source
+	 * @param destSize the number of bytes to be swapped in destination
+	 * 
+	 * @throws IllegalArgumentException if srcOffset is illegal in srcObj, 
+	 * if destOffset is illegal in destObj, or if sizes are invalid
+	 */
+	public void copySwapMemory(Object srcObj, long srcOffset, Object destObj, long destOffset, long srcSize,
+			long destSize) {
+		copySwapMemoryChecks(srcObj, srcOffset, destObj, destOffset, srcSize, destSize);
+
+		if (0 != srcSize) {
+			copySwapMemory0(srcObj, srcOffset, destObj, destOffset, srcSize, destSize);
+		}
+	}
+
+	/**
+	 * Swap specified bytes between source address and destination
+	 * 
+	 * @param srcAddress location to start swap
+	 * @param destAddress location to start swap
+	 * @param srcSize the number of bytes to be swapped at source
+	 * @param destSize the number of bytes to be swapped at destination
+	 * 
+	 * @throws IllegalArgumentException if srcAddress or destAddress is illegal, 
+	 *  or if sizes are invalid
+	 */
+	public void copySwapMemory(long srcAddress, long destAddress, long srcSize, long destSize) {
+		copySwapMemory(null, srcAddress, null, destAddress, srcSize, destSize);
+	}
+
+	/**
+	 * Removes the block from the list. Note that the pointers are located 
+	 * immediately before the startIndex value.
+	 * 
+	 * @param startIndex memory address
+	 * 
+	 * @throws IllegalArgumentException if startIndex is not valid
+	 */
+	public void freeMemory(long startIndex) {
+		freeMemoryChecks(startIndex);
+
+		if (0 != startIndex) {
+			freeMemory0(startIndex);
+		}
+	}
+
+	/**
+	 * Returns byte offset to start of class or interface
+	 * 
+	 * @param field which contains desired class or interface
+	 * @return offset to start of class or interface
+	 * 
+	 * @throws NullPointerException if field parameter is null
+	 * @throws IllegalArgumentException if field is static
+	 */
+	public long objectFieldOffset(Field field) {
+		Objects.requireNonNull(field);
+		return objectFieldOffset0(field);
+	}
+
+	/**
+	 * Returns byte offset to start of static class or interface
+	 * 
+	 * @param field which contains desired class or interface
+	 * @return offset to start of class or iterface
+	 * 
+	 * @throws NullPointerException if field parameter is null
+	 * @throws IllegalArgumentException if field is not static
+	 */
+	public long staticFieldOffset(Field field) {
+		Objects.requireNonNull(field);
+		return staticFieldOffset0(field);
+	}
+
+	/**
+	 * Returns static class or interface
```

Comment:
Its hard to tell what this does from the description. Looking at the native code it returns the declaringClass of the static field.